### DISCUSSION TITLE
Add :not(.vp-isolated) specificity to .vp-doc root elements

### DISCUSSION THREAD

**Discussion by teammate-james** [vuejs/vitepress#3557] [Q&A] [answered]
We are using Vuetify inside our documentation area, but the default `.vp-doc` styles are adversely affecting the UI for some of the components because of root element styles inside `.vp-doc`. For instance, the [default theme](https://github.com/vuejs/vitepress/blob/main/src/client/theme-default/styles/components/vp-doc.css#L110) has a style that begins as `.vp-doc a { ... }`. This ends up applying those styles to ALL `<a>` tags rendered by Vuetify components inside the doc layout causing them to appear improperly. Would it be possible to update the default theme styles by adding another layer of specificity to root element tags which would allow users to isolate from `.vp-doc` styles? For instance, the `.vp-doc a { ... }` CSS could be changed to `.vp-doc :not(.vp-isolated) a { ... }`. That way, if someone wanted to prevent the anchor style from `.vp-doc a` from getting applied to their `<a>` element they could simply wrap their component inside the doc area with a `.vp-isolated` class. The `:not(.vp-isolated)` specifier could be added to all `.vp-doc` styles that are applied to root elements.

**Comment by brc-dd** [comment, marked as answer]
https://vitepress.dev/guide/markdown#raw

**Comment by teammate-james** [comment]
This looks like what we need. I will close this discussion. Thank you.
